### ISSUE TITLE
Set bind params before Query::parse()

### ISSUE THREAD

**Issue by alexbusu** [phalcon/cphalcon#13603]
### Expected and Actual Behavior

```php
$phql = sprintf(
    'SELECT event.id FROM [%s] as event WHERE event.campaign_id in ' .
    '(SELECT campaign.id FROM [%s] campaign WHERE advertiser_id in ({aid:array}))',
    CampaignEvents::class,
    Campaigns::class
);
$modelsManager
    ->createQuery($phql)
    ->execute(['aid' => $args['campaignsOfAdvertiser']]);
```
Exception is thrown with `Bind value is required for array type placeholder: aid` message.
This would work just fine if `::execute()` params would be [set](https://github.com/phalcon/cphalcon/blob/master/phalcon/mvc/model/query.zep#L3498) before [Query::parse()](https://github.com/phalcon/cphalcon/blob/master/phalcon/mvc/model/query.zep#L3371) (and remove params merge block). 
Otherwise I don't see the point to pass bindParams as `::execute()` arguments, especially is the wanted result can be achieved by calling `::setBindParams()` right before execute.
```php
$modelsManager->createQuery($phql)->setBindParams(['aid' => $args['campaignsOfAdvertiser']])->execute();
```

### Details

* Phalcon version: 3.3.2
* PHP Version: 7.2.6

**Comment by CameronHall** [comment]
Hi @alexbusu,

I think you'll find that this has been addressed in 3.4.x branch and you'll see it when 3.4.2 is released. 
https://github.com/phalcon/cphalcon/commit/94f9bb8fa1e271929d8c8a5a2c3d99705523d0ac#diff-b7192acccd6e54692898ab91795e252d

**Comment by alexbusu** [comment]
@CameronHall You're right. I failed to find the existing reported issues, maybe because I was looking for open issues. 
This issue can be closed as it is no longer valid.
